Tests/ComposableArchitectureTests/StoreTests.swift

```diff
@@ -447,4 +447,26 @@ final class StoreTests: XCTestCase {
         .child(2),
       ])
   }
+
+  func testNonMainQueueStore() {
+    for _ in 1...100 {
+      DispatchQueue.global().async {
+        let viewStore = ViewStore(
+          Store.unchecked(
+            initialState: 0,
+            reducer: Reducer<Int, Void, Void>.empty,
+            environment: ()
+          )
+        )
+        viewStore.send(())
+        DispatchQueue.global().asyncAfter(deadline: .now() + 0.1) {
+          viewStore.send(())
+        }
+      }
+    }
+
+    // Allow asychnronous code to run for a bit to make sure the warning breakpoint is
+    // never tripped.
+    _ = XCTWaiter.wait(for: [.init()], timeout: 1)
```

Comment:
I think we are always waiting 1s and failing silently here. Is it the expected behavior?

Reply:
What about:
```swift
func testNonMainQueueStore() {
  var expectations: [XCTestExpectation] = []
  for i in 1 ... 100 {
    let expectation = XCTestExpectation(description: "\(i)th iteration is complete")
    expectations.append(expectation)
    DispatchQueue.global().async {
      let viewStore = ViewStore(
        Store.unchecked(
          initialState: 0,
          reducer: Reducer<Int, Void, Void> {
            state, _, _ in
            state += 1
            if state == 2 { expectation.fulfill() }
            return .none
          },
          environment: ()
        )
      )
      viewStore.send(())
      DispatchQueue.global().asyncAfter(deadline: .now() + 0.1) {
        viewStore.send(())
      }
    }
  }

  wait(for: expectations, timeout: 1)
}
```
On my machine, it runs the test in 0.13s.

Reply:
Great suggestion, thanks!

Reply:
In a true TCA spirit, I now think that the reducer's environment should be the expectation!